### DISCUSSION TITLE
Caddy 2 configuration help

### DISCUSSION THREAD

**Discussion by ilgigante77** [samihsoylu/journal#38] [Q&A] [answered]
Hi , how should I configure Caddy 2 as a https reverse proxy to use Journal?
Any help is welcome!

**Comment by samihsoylu** [comment]
Hi @ilgigante77

Great question, I think this is a little bit tricky, and there are a few articles I found that may help:

* https://php.watch/articles/caddy-php
* https://caddyserver.com/docs/caddyfile/directives/php_fastcgi#expanded-form

Journal comes with a [.htaccess](https://github.com/samihsoylu/journal/blob/master/public/.htaccess) file that has a [rewrite rule for apache2](https://stackoverflow.com/a/6438176). You must take that in to account. Here is a configuration to get you started, it has not been tested (since I do not have a Caddy environment) so you might need to tweak it a little bit.
```
your_domain.com {
    root * /path/to/your/journal/public
    php_fastcgi unix//path/to/php-fpm.sock
    file_server

    @notStatic {
        not file
    }
    rewrite @notStatic /index.php?url={path}&{query}
}
```

Let me know how it goes!

**Reply by ilgigante77** [reply]
Yes this would be correct, I think (another project has a similar configuration), but the Docker container only exposes the port 80 and I don't know how to get the pho-fpm.sock.
In the other project (for the _old but gold_ Flatpress blog) the php container exposes the port 9000 and the html is in a shared volume, so I can put this in the Caddy configuration:
```
root * /var/www/html
php_fastcgi php-fpm:9000
file_server
```
Any ideas?

**Reply by samihsoylu** [reply, marked as answer]
Hi @ilgigante77 

In my initial answer, in the Caddy file I assumed that php-fpm is running on your virtual machine. However, I believe that you might be using the [docker-compose.yml](https://github.com/samihsoylu/journal/blob/master/docker-compose.yml) that comes with Journal. If so, then all you need to do is set up a reverse proxy like this:

```
journal.example.com {
    # The docker compose exposes port 8080
    reverse_proxy localhost:8080
}
```

Feel free to reach out to me on [Discord](https://discord.gg/MzECrZYWHY) if you want to.

**Reply by ilgigante77** [reply]
don't know why it wasn't working the first time I tried the very same caddy instructions, but... they work!
